### PR TITLE
[TTAHUB-372] Add AR creator role

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Added the ability to select a creator role on draft AR's.

## How to test

**Case 1:**
Creating a new report with a user that has only one role should populate the creator role field.

**Case 2:**
If you user has more than one role. The review and submit page of the draft AR should give you the ability to select a role. Once the role is selected it should show in the creator role field.

**Case 3:**
If you user has no roles only the creator name should appear in the creator role field.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-372

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js [hunk 1]
```diff
@@ -58,6 +74,30 @@ const Draft = ({
...
+                  label="Creator role"
+                  name="creatorRole"
+                  required
+                >
+                  <Dropdown
+                    id="creatorRole"
+                    name="creatorRole"
+                    inputRef={register({ required: 'A creator role must be assigned to the report before submitting' })}
+                  >
+                    <option name="default" value="" disabled>- Select -</option>
```
Comment (jasalisbury) [inline frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js:91]: adding `hidden` to this element will remove it from the dropdown

**Review by jasalisbury — COMMENTED** [review]
Appears to be a bug when submitting a report for approval. If I fill out a new report, get to the last page and click `submit report` without clicking `save draft` I get a validation error and the report has not been submitted.
```
backend_1      | 2022-03-02T23:22:18.230Z AUDIT error: 	SERVICE:ACTIVITY_REPORTS id: 1255 Sequelize error SequelizeValidationError: Validation error: Missing required field(s)
```
If I click `save draft` before clicking `submit for approval` I do not get an error.

File: src/constants.js [hunk 2]
```diff
@@ -106,3 +106,21 @@ export const TARGET_POPULATIONS = [
   'Children with Special Health Care Needs',
   'Dual-Language Learners',
 ];
+
```
Comment (thewatermethod) [inline src/constants.js:109]: not for this ticket obv, but I wonder if we should make a package with all the constants shared between frontend and backend.

**Comment by thewatermethod** [comment]
> Creating a new report with a user that has only one role should populate the creator role field.

Created a report for a user with only one role and submitted it. Noticed that the "Creator role" field is empty in the db and the creatorNameWithRole data in the API is only returning the name.

Not sure if I'm misunderstanding something in the workflow?

What the frontend gets back as the report data:

Screen Shot 2022-03-03 at 9 51 30 AM

User info from API:

Screen Shot 2022-03-03 at 9 51 37 AM

DB query for that report

Screen Shot 2022-03-03 at 9 52 02 AM

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/__tests__/index.js [hunk 3]
```diff
@@ -54,26 +59,31 @@ const renderReview = (
...
     calculatedStatus,
     displayId: '1',
     id: 1,
+    creatorRole,
   };
 
   const history = createMemoryHistory();
   const pages = complete ? completePages : incompletePages;
-
+  // console.log('\n\n\nPassed User: ', passedUser.role);
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/__tests__/index.js]: Removed.

**Review by thewatermethod — COMMENTED** [review]
on main right now
- create a new report
- don't fill in any data, click on review
- click back to next steps, no error

on this branch those same steps cause the backend to error out. 

Everything else seems to work pretty good

**Comment by thewatermethod** [comment]
Sorry - one more question

The approved AR view shows all creator roles, should it only show the selected one?

**Comment by thewatermethod** [comment]
Same question for CSV actually

**Comment by AdamAdHocTeam** [comment]
> Sorry - one more question
> 
> The approved AR view shows all creator roles, should it only show the selected one?

Yea its next on the list via [TTAHUB-693](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-693).

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**